core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/api/update.kt

```diff
@@ -40,10 +46,45 @@ internal fun <T, C> Update<T, C>.updateWithValuePerColumnImpl(selector: Selector
         }
     }
 
+/**
+ * Implementation for Update As Frame:
+ * Replaces selected column groups with the result of the expression only where the filter is true.
+ */
+internal fun <T, C, R> Update<T, DataRow<C>>.asFrameImpl(expression: DataFrameExpression<C, DataFrame<R>>): DataFrame<T> =
+    if (df.isEmpty()) df
+    else df.replace(columns).with {
+        val src = it.asColumnGroup()
+        val updatedColumn = expression(src, src).asColumnGroup(src.name())
+        if (filter == null) {
```

Comment:
What's happened if we will have some other operations here

Reply:
let me think, I don't remember all the scope of operations

Reply:
I do, I just kdoc'ed the entirety of update.kt XD